lib/gui/app.js

```diff
@@ -180,17 +184,15 @@ app.run(() => {
     //
     // We use the presence of `.eta` to determine that the actual
     // writing started.
-    if (!flashState.isFlashing() || !currentFlashState.eta) {
-      return
+    if (!currentFlashState.eta) {
+      analytics.logDebug([
+        `Progress (${currentFlashState.type}):`,
+        `${currentFlashState.percentage}% at ${currentFlashState.speed} MB/s`,
+        `(eta ${currentFlashState.eta}s)`
+      ].join(' '))
```

Comment:
Can we change this to use string concatenation, so we don't allocate a new array on every call?

Reply:
Done this now!